### PR TITLE
project todo - Noemi

### PR DESCRIPTION
**Pull request by NoemiSalome**
Link to my deployed project: [https://project-todo-noemi.netlify.app/]

My reflections on how this week's project turned out: a lot of information and still trying to put everything into its place. discovered styling with styled components, which is great fun! would have loved to implement a button to delete all tasks, but couldn't make it work. Looking forward to some more redux :)

Things I'd like to have clarified or explained in more detail:
styling: somehow when submiting a task the container shrinks and the tasks don't take the width of the whole container. I tried putting width = 100%, but didn't work. Would love some input on that. spent hours trying to solve it. THANKS 😊

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+This weeks project was about pulling together all the parts of redux und building it's structure from scratch.
 
 ## The problem
-
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+First of all I started by sketching the app on paper. Mondays lecture already gave a good idea how to start and set up everything. I had (still have) a bit of a hard time wraping my head around the whole thing. The concept is hard to grasp for me and at some point I just had to accept that I don't fully understand what's going on in my code otherwise I wouldn't be able to hand it in. 
+This week I wanted to work with styled components and as it turns out I prefer it this way. It makes the styling process more structured and quicker. I found all the answers in the recordings of the lectures, on stackoverflow and google. Also thanks to youtube I was able to implement a lottie animation. 
 
 ## View it live
```
Comment (MarySparrow) [inline README.md]: It's a great way to start your project sketching it first, and it turned out very nice and pretty! Be proud! I would love to see that video on how to do the Lottie adding, way to go to get that setup! 😍🤩 
Some Lottie envy going on over here 🤠🤠

File: code/src/reducers/tasks.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                    }
+                } else {
+                    return task
+                }
+            })
+            store.items = updatedItems
+        },
+        removeTask: (store, action) => {
+            store.items.splice(action.payload, 1)
+        },
```
Comment (MarySparrow) [inline code/src/reducers/tasks.js:24]: There is a little issue with the deleting of a task, it removes the first item of the list not the one you click. This has to do with a mutable approach vs immutability. I cannot say why it is not working but I can show you how the immutability approach can be written and you can try that instead. Maybe you need to write in more than one place, a little help for that please code coach 💁 

```
removeTasTk: (store, action) => {
      const removeTask = store.tasks.filter(todo => todo.id !== action.payload) 
        store.tasks = removeTask
    }
```

File: code/src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    const items = useSelector((store) => store.tasks.items)
+    const completedTasks = items.filter(item => item.isComplete)
+
+    return (
+        <HeaderContainer>
+            <div>
+                <Titel>Todo</Titel>
+                <Normaltext>
+                    {moment(new Date().toJSON()).format('MMMM Do YYYY')}
+                </Normaltext>
```
Comment (MarySparrow) [inline code/src/components/Header.js]: Good job in getting the time , cred for the extra effort! --> a little note though, for future projects (or whenever we have the time to go back to fix all our piling up stuff) , moment is no longer being maintained and so there is a high risk of bugs and so on.. 🐞 🐞

File: code/src/components/TaskList.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: space-between;
+    margin-top: 4px;
+    margin-bottom: 8px;
+    border-bottom: 1px solid black;
+    word-break: break-word
+`
+
```
Comment (MarySparrow) [inline code/src/components/TaskList.js]: A wrote the longest word ever and it worked like a charm, must be this setting you remembered right here 👍

File: code/src/components/TaskList.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    height: 13px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin: 3px 4px 3px 3px;
+    background-color: #E5E5E5;
+    border: 1px solid grey;
+    border-radius: 3px
+`
+
```
Comment (MarySparrow) [inline code/src/components/TaskList.js:44]: I know you said you were gonna work with styled component for the first time (i think) this week and it turned out great! ⭐  I also prefer that now, lets find out the advanced stuff too, React for life haha! 😊😊

File: code/src/components/TaskList.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                                checked={task.isComplete}
+                                onChange={() => dispatch(tasks.actions.toggleComplete(task.id))}
+                            />
+                            <Normaltext>{task.description}</Normaltext>
+                            <DeleteButton 
+                                onClick={() => dispatch(tasks.actions.removeTask(task.id))}>
+                                <ButtonText>X</ButtonText>
+                            </DeleteButton>
+                        </SingleTaskContainer>
+                    ))}
```
Comment (MarySparrow) [inline code/src/components/TaskList.js]: very clear namings ! 👌

File: code/src/components/TaskList.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import { useSelector, useDispatch } from 'react-redux'
+import styled from 'styled-components/macro'
+
+import tasks from '../reducers/tasks'
+import EmptyStateAnimation from './EmptyStateAnimation'
+
+const TaskContainer = styled.div`
+    overflow: auto;
+    width: 100%;
+`
```
Comment (MarySparrow) [inline code/src/components/TaskList.js]: Maybe add a max-width here, the textinput field get a bit large on the laptop 💻 (and I know I messed your styling in the last minute yesterday haha, sooorrryyy!) 🙇‍♀️

File: code/src/App.js [hunk 8]
```diff
@@ -1,9 +1,39 @@
...
+  width: 100%;
+  justify-content: space-between;
+    @media (min-width: 768px) {
+      height: 80vh;
+      width: 50vw;
+    } 
+    @media (min-width: 1200px) {
+      width: 50vw;
+    } 
+`
```
Comment (MarySparrow) [inline code/src/App.js]: You always remember those media Qs 😺😺 good job!

File: code/src/components/Header.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+const Titel = styled.h1`
+    font-size: 20px;
+`
+
+const Normaltext = styled.p`
+    font-size: 12px;
+`
+
+const Header = () => {
+    const items = useSelector((store) => store.tasks.items)
```
Comment (MarySparrow) [inline code/src/components/Header.js]: try to keep the same pattern of space between styled components 👽

**Review by MarySparrow — CHANGES_REQUESTED** [review]
Awesome job Noemi, I always enjoy your projects, I still cant believe you succeed to hand them in in time and completed when starting sometimes very late in the week. You have some major skills and high stress level girl! 💃  and on top of that always have time for a break and talk about my CSS issues 😂

I hope I get to work with you on another project , that would be truly fun! 👯 

✔️Just fix that little delete button and you are minutes from another completed project ✔️

File: code/src/reducers/tasks.js (hunk 2 above)
Comment (KarolCislowski) [reply code/src/reducers/tasks.js:24]: Yes @MarySparrow is 100% right 🥇

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Nice work but you need to fix the `remove task` issue to meet blue level requirments. 

Your tasks are taking 100% of container width, so I don't know why they work differently on your machine. 
image

**Review by KarolCislowski — APPROVED** [review]
Great work!

**Pull request closed without merging**